src/main/java/pl/marekjava/PasswordManager.java

```diff
@@ -14,12 +13,38 @@ public PasswordManager() {
         passwords.add("Darowanemu koniowi w zęby się nie zagląda");
         passwords.add("Diabeł tkwi w szczegółach");
         passwords.add("Elektryka prąd nie tyka");
+
+        for (int i = 0; i < passwords.size(); i++) {
+            usedPasswords.put(passwords.get(i), false);
+        }
     }
 
     public String getRandomPassword() {
         Random r = new Random();
         int randomItem = r.nextInt(passwords.size());
         String randomPassword = passwords.get(randomItem);
+
+        int counter = 0;
+        while (usedPasswords.get(randomPassword) == true) {
```

Comment:
tutaj wystarczy `while(usedPasswords.get(randomPassword))`, bo to już zwróci Boolean'a, który potem jest przekształcony na boolean, a finalnie na true/false, więc porównanie do true jest już niepotrzebne.